chore: add process, skripts and actions

morri-son opened the pull request:
On-behalf-of: Gerald Morrison (SAP) <gerald.morrison@sap.com>

<!-- markdownlint-disable MD041 -->
#### What this PR does / why we need it
Create versioning process, scripts and GH actions

morri-son commented:
Main. part of the PR is the process description for content creation and how to do the versioning. The scripts and the GH actions can be used later to automate it. Currently they need to be triggered manually and most-likely need to be adopted later after we tried-out the process and refined it (or not :-)).

jakobmoellerdev reviewed (commented):
Was this process already tried by you? For me it looks like we have dead scripts that we should only check in once we actually have automation enabled. Right now the PR is in. abit of a weird spot for me. WDYT?

fabianburth commented:
Without having tried it to potentially discover some gaps in the description, the process description looks rather thorough. Though, at the moment, this would be a complete blackbox for me. I would have no clue about why these steps are required.

morri-son commented on .github/config/wordlist.txt:
```diff
@@ -241,6 +241,7 @@ mvn
 mvnrepository
 namespace
 namespaces
+neonephos
```
not required as spellchecker's wordlist does not take caps into account.

morri-son commented:
> Was this process already tried by you? For me it looks like we have dead scripts that we should only check in once we actually have automation enabled. Right now the PR is in. abit of a weird spot for me. WDYT?

The process has been tested manually, but only for new release tags. I identified one thing that is missing: update Hugo modules that keep having the same version tag, in our case "main" from the open-component-model repo for the CLI ref docs. I need to check how to force updates for the same version tag.

I removed the scripts and action from this PR and moved them to another branch and will open a new PR once we want to automate the process.

morri-son commented:
I've split this PR into "docs creation" and "versioning process", therefore closing this one

https://github.com/open-component-model/ocm-website/pull/536
https://github.com/open-component-model/ocm-website/pull/537
